### ISSUE TITLE
Risk of hitting database connection limit

### ISSUE THREAD

**Issue by relrod** [ansible/tower-operator#1151]
As @TheRealHaoLiu predicted, with the web/task split work, as we introduce new tooling that relies on connection to postgres, we consume more and more connections to the database.

Right now the default postgres.conf in our postgres container has max_connections set to 100.

With 4 task replicas and 26 web replicas, I see this start showing up in awx-task containers (this might trigger with less, I haven't tested to see when it starts):

```
awx-task-67874cfd5d-fckk7 awx-task 2022-12-11 06:41:06,978 ERROR    [dfa3c99a] awx.main.dispatch.periodic encountered an error while scheduling periodic tasks
awx-task-67874cfd5d-fckk7 awx-task Traceback (most recent call last):
awx-task-67874cfd5d-fckk7 awx-task   File "/awx_devel/awx/main/dispatch/periodic.py", line 37, in run
awx-task-67874cfd5d-fckk7 awx-task     self.run_pending()
awx-task-67874cfd5d-fckk7 awx-task   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/schedule/__init__.py", line 94, in run_pending
awx-task-67874cfd5d-fckk7 awx-task     self._run_job(job)
awx-task-67874cfd5d-fckk7 awx-task   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/schedule/__init__.py", line 147, in _run_job
awx-task-67874cfd5d-fckk7 awx-task     ret = job.run()
awx-task-67874cfd5d-fckk7 awx-task   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/schedule/__init__.py", line 466, in run
awx-task-67874cfd5d-fckk7 awx-task     ret = self.job_func()
awx-task-67874cfd5d-fckk7 awx-task   File "/awx_devel/awx/main/dispatch/publish.py", line 96, in apply_async
awx-task-67874cfd5d-fckk7 awx-task     with pg_bus_conn() as conn:
awx-task-67874cfd5d-fckk7 awx-task   File "/usr/lib64/python3.9/contextlib.py", line 119, in __enter__
awx-task-67874cfd5d-fckk7 awx-task     return next(self.gen)
awx-task-67874cfd5d-fckk7 awx-task   File "/awx_devel/awx/main/dispatch/__init__.py", line 71, in pg_bus_conn
awx-task-67874cfd5d-fckk7 awx-task     pg_connection.connect()
awx-task-67874cfd5d-fckk7 awx-task   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/utils/asyncio.py", line 33, in inner
awx-task-67874cfd5d-fckk7 awx-task     return func(*args, **kwargs)
awx-task-67874cfd5d-fckk7 awx-task   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/backends/base/base.py", line 200, in connect
awx-task-67874cfd5d-fckk7 awx-task     self.connection = self.get_new_connection(conn_params)
awx-task-67874cfd5d-fckk7 awx-task   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/utils/asyncio.py", line 33, in inner
awx-task-67874cfd5d-fckk7 awx-task     return func(*args, **kwargs)
awx-task-67874cfd5d-fckk7 awx-task   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/backends/postgresql/base.py", line 187, in get_new_connection
awx-task-67874cfd5d-fckk7 awx-task     connection = Database.connect(**conn_params)
awx-task-67874cfd5d-fckk7 awx-task   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/psycopg2/__init__.py", line 122, in connect
awx-task-67874cfd5d-fckk7 awx-task     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
awx-task-67874cfd5d-fckk7 awx-task psycopg2.OperationalError: FATAL:  sorry, too many clients already
```

We might need perf/scale input here - can we just increase the max_connections that postgres allows?

CCs:
perf/scale- @kdelee @rebeccahhh
operator- @rooftopcellist @shanemcd

**Comment by TheRealHaoLiu** [comment]
Question… in original configuration (before split) how many db connection per replica do we establish statically (without running jobs)

**Comment by kdelee** [comment]
This is been a thorn in our side, downstream testing overrides `postgres_extra_args` to up the max connections. One pain is that upstream/downstream postgres images require different formulation of the extra args.

Default for VM based cluster (e.g. more than 1 control node) is 1024 connections, in general it is poor housekeeping on our part not to scale max connections with the size of the control plane. **Ideally, max_connections should always be greater than max uwsgi workers + max concurrent system tasks + max job tasks actually using a DB connection.** 

We know jobs use a database connection on start, and we have this START_TASK_LIMIT setting.

Right now max uwsgi workers is 16*number of control plane nodes/pods -- maybe would change with the web/task split.

Changing postgres max connections config isn't something I think we can't do on the fly (see https://www.postgresql.org/message-id/3393.1156528520%40sss.pgh.pa.us ) so arbitratily scaling the web workers without keeping max db connections in mind is a good way to have a bad time.

tl;dr:

100 is a bad default for production systems, probably good for CI jobs that use very mini-deployment, downstream testing overrides this and we've had customer bugs around this  (cc @rooftopcellist ).

**Comment by thedoubl3j** [comment]
Not an issue that was introduced by the web task split feature but something that we need to investigate as we go forward.

**Comment by TheRealHaoLiu** [comment]
number of static db connection increase
for web 6 connection
- 2 for cache clearer
- 2 for rsyslog configure
- 2 for heartbeet
for task 2 additional connection
- 2 for rsyslog configure
